### PR TITLE
Added create endpoint 

### PR DESCRIPTION
**Pull request by madams51703**
I want to add in some labels, but had to remove them to get the merge conflicts resolved.  Don't really understand why that generated a conflict, but for now I removed them.  After this merges, I want to add the labels back in and try another pull request.  I appreciate the patience with me as I learn more about git.

### CODE REVIEW

**Comment by tuxmania87** [comment]
Could you also remove all the trailing whitespace you added?

**Comment by KvanTTT** [comment]
* Fix grammar for [test](https://travis-ci.org/antlr/grammars-v4/builds/255915565#L4728) passing.
* Add code fragment to test if possible.

**Comment by madams51703** [comment]
I removed whitespace from function_call rules, added example files and fixed the errors I introduced to the grammar.  Please let me know if I missed anything or if there is anything else I need to do.  Thanks again for your patience.

**Comment by KvanTTT** [comment]
Travis [still failing](https://travis-ci.org/antlr/grammars-v4/builds/256026924#L4759).

**Comment by madams51703** [comment]
I will investigate further after work.

**Comment by madams51703** [comment]
Ok
 for i in examples/*.sql; do echo ${i}; cat ${i} | tr [:lower:] [:upper:] | grun TSql tsql_file ; done
examples/analytic_windowed_functions.sql
examples/control_flow.sql
examples/cursors.sql
examples/dbcc.sql
examples/ddl_alter_database_mirroring.sql
examples/ddl_create_alter_database.sql
examples/ddl_create_drop_type.sql
examples/ddl_create_endpoint.sql
examples/ddl_create_table.sql
examples/ddl_function.sql
examples/ddl_index.sql
examples/ddl_table.sql
examples/dml_delete.sql
examples/dml_insert.sql
examples/dml_merge.sql
examples/dml_openrowset.sql
examples/dml_select.sql
examples/dml_update.sql
examples/expressions.sql
examples/full_width_chars.sql
examples/predicates.sql
examples/procedures.sql
examples/statements.sql
examples/transactions.sql
examples/triggers.sql
examples/xml_data_type.sql
results of my testing against the examples

**Comment by madams51703** [comment]
I also added back the labels for the functions.  Those can be very useful in the future.

**Comment by teverett** [comment]
@KvanTTT all good w/ this one?

File: tsql/TSqlParser.g4 [hunk 1]
```diff
@@ -1411,52 +1550,52 @@ derived_table
...
+    | SESSION_USER					#SESSION_USER
     // https://msdn.microsoft.com/en-us/library/ms179930.aspx
-    | SYSTEM_USER
+    | SYSTEM_USER					#SYSTEM_USER
     // https://msdn.microsoft.com/en-us/library/ms184325.aspx
-    | ISNULL '(' expression ',' expression ')'
+    | ISNULL '(' expression ',' expression ')'		#ISNULL
     // https://docs.microsoft.com/en-us/sql/t-sql/xml/xml-data-type-methods
-    | xml_data_type_methods
+    | xml_data_type_methods				#XML_DATA_TYPE_METHODS
```
Comment (KvanTTT) [inline tsql/TSqlParser.g4]: I'm not sure about alternative labels here. @Muhamedali what do you say about that?

**Comment by madams51703** [comment]
The reason for the labels is to be able to directly manipulate those
specific rules and provide translations for those functions

**Comment by teverett** [comment]
@madams51703 @KvanTTT so, good to go?

**Comment by madams51703** [comment]
@teverett  I am good to go.  But I am biased.  :+)

**Comment by KvanTTT** [comment]
Overall ok :+1:

**Comment by madams51703** [comment]
Oh, @KvanTTT  I read your comment from a different perspective.  If you specifically are referring to the xml_data_type_methods line, adding the labels is an all or nothing situation and I had no choice but to add it to the xml_data_type_methods line if I wanted to add it to the other functions in that ruleset.

**Pull request merged**